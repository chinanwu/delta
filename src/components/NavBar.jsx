import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import getThemeClassname from '../functions/getThemeClassname';

import './NavBar.less';

const getClassName = (tab, active, dark) =>
	getThemeClassname(
		tab === active ? 'NavBar__btn--disabled' : 'NavBar__btn',
		dark
	);

export const NavBar = ({ activeTab, dark }) => (
	<div className="NavBar" role="navigation" aria-label="Game">
		<Link className={getClassName('home', activeTab, dark)} to="/">
			Home
		</Link>
		<Link className={getClassName('about', activeTab, dark)} to="/about">
			About
		</Link>
		<Link className={getClassName('settings', activeTab, dark)} to="/settings">
			Settings
		</Link>
	</div>
);

NavBar.propTypes = {
	activeTab: PropTypes.string,
	dark: PropTypes.bool,
};

export const mapStateToProps = ({ theme }) => ({
	dark: theme.dark,
});

export default connect(mapStateToProps)(NavBar);
