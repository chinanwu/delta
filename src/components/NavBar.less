@import '../style/styles';

.NavBar {
	@nav();
}

.NavBar__btn {
	@navBtn();
}

.NavBar__btn--dark {
	@darkNavBtn();
}

.NavBar__btn:hover {
	@navBtnHover();
}

.NavBar__btn--dark:hover {
	@darkNavBtnHover();
}

.NavBar__btn:active {
	@navBtnActive();
}

.NavBar__btn--dark:active {
	@darkNavBtn();
}

.NavBar__btn--disabled {
	@disabledNavBtn();
}

.NavBar__btn--disabled--dark {
	@disabledDarkNavBtn();
}
