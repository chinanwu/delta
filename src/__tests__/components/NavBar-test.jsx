import { shallow } from 'enzyme';
import React from 'react';

import { NavBar, mapStateToProps } from '../../components/NavBar.jsx';

jest.unmock('../../components/NavBar.jsx');

describe('NavBar component', () => {
	describe('rendering', () => {
		it('renders NavBar', () => {
			const wrapper = shallow(<NavBar />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});
	});

	describe('mapStateToProps', () => {
		it('returned mapped properties', () => {
			const theme = { dark: true };
			const state = { theme };
			expect(mapStateToProps(state)).toEqual(theme);
		});
	});
});
