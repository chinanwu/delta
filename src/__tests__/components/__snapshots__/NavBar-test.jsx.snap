// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component rendering renders NavBar 1`] = `
<div
  aria-label="Game"
  className="NavBar"
  role="navigation"
>
  <ForwardRef
    to="/"
  >
    Home
  </ForwardRef>
  <ForwardRef
    to="/about"
  >
    About
  </ForwardRef>
  <ForwardRef
    to="/settings"
  >
    Settings
  </ForwardRef>
</div>
`;
